File: backend/src/main/java/com/example/stamp_app/domain/common/entity/AppResponseObject.java
package com.example.stamp_app.domain.common.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Getter
@NoArgsConstructor
public class AppResponseObject {

    Object response;

    List<Error> errors = new ArrayList<>();

    public static AppResponseObject createErrorResponse(String code, String message) {
        var response = new AppResponseObject();
        var error = new Error(code, message);
        response.addError(error);
        return response;
    }

    public void addError(Error error) {

        this.errors.add(error);

    }

}

File: backend/src/main/java/com/example/stamp_app/domain/common/entity/AppResponseObject.kt
package com.example.stamp_app.domain.common.entity

data class AppResponseObject(
	private val response: Any? = null,
	private val errors: MutableList<Error> = ArrayList()
) {

	fun addError(error: Error) {
		errors.add(error)
	}

	companion object {

		/**
		 * エラーを含んだレスポンスObjectを返す
		 *
		 * @param code エラーコード
		 * @param message エラーメッセージ
		 */
		fun createErrorResponse(code: String?, message: String?): AppResponseObject {
			val response = AppResponseObject()
			val error = Error(code, message)
			response.addError(error)
			return response
		}

	}
}
